app/assets/javascripts/templates/home/index.html.haml

```diff
@@ -1,6 +1,23 @@
 .home-container{ ng_controller: 'homeCtrl' }
-  %h1
-    Recipe
-  %hr
-    %p{ ng_show:  "signedIn()" }
-      Welcome {{ user.email }}
+
+  %p{ ng_show:  "signedIn()" }
+    Welcome {{ user.email }}
+
+  #search
+    %label{for: 'search-input'}
+      %i{ class: 'fa fa-search'}
+    %input{ id: 'search-input' , ng_model: 'searchRecipe', class: 'form-control  search-query',placeholder:'Search'}
+    %a{ id: 'search-clear', class:'fa fa-times-circle', tabindex:'1' , ng_show:'searchRecipe' , ng_click:'searchRecipe = null'}
+
+  #recipes
+    %div{ ng_repeat: 'r in recipes'}
```

Comment:
please never and ever use a one-letter variables it can be a conflict source 

` %div{ ng_repeat: 'recipe in recipes' } `